[#278] Search works by indexTime and use operators

ponchofiesta opened the pull request.

pbroman commented on kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/Operator.java:
```diff
@@ -0,0 +1,53 @@
+    EQUAL("="),
+    NOTEQUAL("!="),
+    CONTAINS(":"),
+    NOTCONTAINS("!:")
```
This is really a small matter, but the convention for naming constants in java is with underscore, i.e. GREATER_OR_EQUAL etc., and so it would be easier to read.

ponchofiesta commented:
OK, that data pattern can not be parsed by LocalDateTime. I removed it. Another problem is, that I would expect indexTime="2019-01-01" would match to any DateTime with that date and the time part should be ignored. For the same reason I removed "today".

And now NULL is handled in other operators too.

And the operator constants are renamed.
